### ISSUE TITLE
is Flash attention2 supprted?

### ISSUE THREAD

**Issue by rajveer43** [pytorch-labs/gpt-fast#18]

**Comment by MF-FOOM** [comment]
https://github.com/pytorch-labs/gpt-fast/blob/3bcaaaf068d112d534f335ec21a17d7b8b5551bf/model.py#L182C15-L182C43

By using `scaled_dot_product_attention`, they're using Flash Attention 2, as `scaled_dot_product_attention` uses Flash Attention 2 by default now (https://github.com/pytorch/pytorch/pull/105602)

**Comment by Chillee** [comment]
Yes what @MF-FOOM said. In addition, FA2 is not beneficial during the decoding phase, so it's actually more beneficial for us to codegen it from scratch. It would be faster to use the FlashDecode kernel, but that's not fully integrated into PyTorch yet.
